modules/upload_contacts_from_file.py

```diff
@@ -0,0 +1,121 @@
+import csv
+import re
+from modules import utils, visit_date_helper
+from management.models import Contact
+
+def csv_upload(filepath):
+	with open(filepath) as csvfile:
+		reader = csv.DictReader(csvfile)
+		
+		for row in reader:
+			new_dict = make_contact_dict(row)
+			
+			try:
+				new_contact = Contact.objects.get(name=new_dict["name"],phone_number=new_dict["phone_number"])
+			except Contact.DoesNotExist:
+				new_contact = Contact.objects.get_or_create(**new_dict)
```

Comment:
It does. So why do this?  I don't want to test existing contacts against the complete dict that is being uploaded (may be old contacts with new info) but rather a few categories (and checking for an id would require tons of non-database work ensuring csv files and existing contacts don't have duplicates or potentially a model change).

So, that means a Contact needs to attempted to be fetched via `get`, `get_or_create`, or possibly `update_or_create` with the select parameters and then whether or not such a contact exists the fields have to be updated with a call to `Contact.objects.get_or_create(**new_dict)`. Since two calls are being made either way, I found this a clear version of what was happening, though creating a new Contact with `create()` and `save()` if it didn't exist could have also worked.